### ISSUE TITLE
toml parser rejects array of table syntax following a non-empty inline table

### ISSUE THREAD

**Issue by wetheredge** [oven-sh/bun#13230]
### What version of Bun is running?

1.1.22-canary.96+df33f2b2a

### What platform is your computer?

Linux 6.9.8-200.fc40.x86_64 x86_64 unknown

### What steps can reproduce the bug?

Attempt to import a toml file that contains an inline table with at least 1 key followed by an array of tables.

For example, `bun run ./error.js` with these two files:

```javascript
// error.js
import data from './error.toml';
console.log(data);
```

```toml
# error.toml
bug = { not = "empty" }
# no-bug = {}
# [no-bug]
# no-bug = true
[[array]]
```

Uncommenting any of the `no-bug` lines resolves the issue.

### What is the expected behavior?

```javascript
{
  bug: {
    not: "empty",
  },
  array: [
    {}
  ],
}
```

### What do you see instead?

```
5 | [[array]]
     ^
error: Expected key but found [
    at /tmp/error.toml:5:2

Bun v1.1.22-canary.96+df33f2b2a (Linux x64)
```

**Comment by Hamza12700** [comment]
I tested it on bun `1.1.23 canary build` and still doesn't work. It's likely a bug in the parser then.

**Comment by wetheredge** [comment]
Agreed. My guess is that something is setting `p.lexer.has_newline_before = false` and not putting it back when it should. Perhaps while handling `.t_open_brace`?

**Comment by nektro** [comment]
fixed in bun@1.2.16
